webapp/src/components/CrewAppForm.tsx

```diff
@@ -115,6 +122,33 @@ export default function CrewAppForm({
 		initialTasks || []
 	);
 
+	const taskIDs = tasksState?.map(t => t.value);
+	const selectedTasks = taskIDs?.map(t => taskChoices.find(tc => tc._id === t));
+	const variableIdsOfSelectedTasks = selectedTasks?.map(t => t?.variableIds || []).flat();
+	const variablesOfSelectedTasks = variableIdsOfSelectedTasks?.map(v =>
+		variableChoices?.find(vc => vc._id === v)
+	);
+
+	const agentIds = agentsState?.map(a => a.value);
+	const selectedAgents = agentIds?.map(a => agentChoices.find(ac => ac._id === a));
+	const variableIdsOfSelectedAgents = selectedAgents?.map(a => a?.variableIds || []).flat();
+	const variablesOfSelectedAgents = variableIdsOfSelectedAgents?.map(v =>
+		variableChoices?.find(vc => vc._id === v)
```

Comment:
The variable may not be found, suggest to filter this list for safety but also handle this in backend when saving tasks, agents, etc